### PR TITLE
feat(mongodb-datastore): Use cp-connector library

### PR DESCRIPTION
**Pull request by RealAnna**
## This PR

- adds cp connector instead of distributor
- adds unit tests
- refactor a bit swagger file to use common keptn event model from go utils

### Related Issues

#7435 

https://github.com/keptn/keptn/runs/6386309910?check_suite_focus=true

### CODE REVIEW

File: installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml [hunk 1]
```diff
@@ -86,25 +86,10 @@ spec:
...
-          - name: PUBSUB_IMPL
-            value: nats
-          - name: PUBSUB_TOPIC
-            value: 'sh.keptn.>'
-          - name: PUBSUB_RECIPIENT
-            value: '127.0.0.1'
-          - name: PUBSUB_RECIPIENT_PATH
-            value: '/event'
-          {{- include "control-plane.dist.common.env.vars" . | nindent 10 }}
+        - name: PUBSUB_IMPL
```
Comment (odubajDT) [inline installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml:89]: Do we actually need this variable? It isn't used anywhere

File: mongodb-datastore/handlers/config.go [hunk 2]
```diff
@@ -2,13 +2,41 @@ package handlers
...
 )
 
 const (
 	eventsCollectionName = "keptnUnmappedEvents"
 	serviceName          = "mongodb-datastore"
 )
 
+type envConfig struct {
+	// Port on which to listen for cloudevents
+	Port                    int    `envconfig:"RCV_PORT" default:"8080"`
```
Comment (odubajDT) [inline mongodb-datastore/handlers/config.go:15]: Port and Path not used, please remove them. Also most of the other env variables are not set in helm charts, please use this to set them up https://github.com/keptn/keptn/commit/bceaf4b5e298170073c566fc28361ae474e80898#diff-3cd7b0431852a6177235bd58cfc84fcbde3a836c2098a1d1bd5fab2387b8c8faR86

Example here: https://github.com/keptn/keptn/commit/bceaf4b5e298170073c566fc28361ae474e80898#diff-ded52bfd23c5a4a266c21154c749dddc94c40b6dc2c7b9fff34b455dbf87c6c5R68

Please also adapt deploy/service.yaml accordingly

File: installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml (hunk 1 above)
Comment (RealAnna) [reply installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml:89]: yeah this was used by distributor to decide between HTTP and nats but indeed not needed

**Review by odubajDT — APPROVED** [review]
lgtm, but let's wait for the integration tests to finish :)

**Review by odubajDT — APPROVED** [review]

**Pull request merged**